### PR TITLE
fix: Add custom style hooks to MessageBar components

### LINKED ISSUE #29429: [Bug]: MessageBar does not support custom style override
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
v9 MessageBar preview
```

### Are you reporting Accessibility issue?

None

### Reproduction

docsite

## Actual Behavior
No `useCustomStyleHook_unstable` in components in MessageBar

## Expected Behavior
MessageBar components should support `useCustomStyleHook_unstable`

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by ling1726**
Fixes #29429

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                          |        Size (minified/GZIP) |
| ---------------------------------------------------------------------------------------------------------- | --------------------------: |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr> | `25.305 kB`<br />`9.162 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/60a911e66db784521a35c1e1ac0362a3f2d8169f'>60a911e66db784521a35c1e1ac0362a3f2d8169f</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">73</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/InfoButton-mount_base.html">12</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/InfoButton-mount.html">14</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/Avatar-mount_base.html">619</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/Avatar-mount.html">618</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/Button-mount_base.html">314</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/Button-mount.html">322</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/Field-mount_base.html">1158</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/Field-mount.html">1152</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProvider-mount_base.html">703</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProvider-mount.html">691</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProviderWithTheme-mount_base.html">81</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProviderWithTheme-mount.html">86</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">68</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">71</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">73</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/InfoButton-mount_base.html">12</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/InfoButton-mount.html">14</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/MakeStyles-mount_base.html">851</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/MakeStyles-mount.html">857</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/Persona-mount_base.html">1744</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/Persona-mount.html">1680</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/SpinButton-mount_base.html">1404</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29431/perf-test-react-components/SpinButton-mount.html">1363</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: change/@fluentui-react-message-bar-preview-61fbedf2-6e27-44a3-9ab7-a8e3eac0c3c9.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "https://github.com/microsoft/fluentui/issues/29408",
```
Comment (sopranopillow) [inline change/@fluentui-react-message-bar-preview-61fbedf2-6e27-44a3-9ab7-a8e3eac0c3c9.json:3]:
```suggestion
  "comment": "feat: Add custom style hook to MessageBar.",
```

**Review by sopranopillow — APPROVED** [review]

**Review by layershifter — APPROVED** [review]

**Pull request merged**